internal/cltest/cltest.go

```diff
@@ -613,16 +613,25 @@ func CreateJobRunViaWeb(t *testing.T, app *TestApplication, j models.JobSpec, bo
 
 // CreateHelloWorldJobViaWeb creates a HelloWorld JobSpec with the given MockServer Url
 func CreateHelloWorldJobViaWeb(t *testing.T, app *TestApplication, url string) models.JobSpec {
-	j := FixtureCreateJobViaWeb(t, app, "../internal/fixtures/web/hello_world_job.json")
-	j.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
-	return CreateJobSpecViaWeb(t, app, j)
+	buffer, err := ioutil.ReadFile("../internal/fixtures/web/hello_world_job.json")
+	if err != nil {
+		assert.FailNowf(t, "Unable to read fixture", err.Error())
+	}
+
+	var job models.JobSpec
+	err = json.Unmarshal(buffer, &job)
+	require.NoError(t, err)
+
+	job.ID = utils.NewBytes32ID()
+	job.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
+	return CreateJobSpecViaWeb(t, app, job)
 }
 
 // CreateMockAssignmentViaWeb creates a JobSpec with the given MockServer Url
 func CreateMockAssignmentViaWeb(t *testing.T, app *TestApplication, url string) models.JobSpec {
 	j := FixtureCreateJobWithAssignmentViaWeb(t, app, "../internal/fixtures/web/v1_format_job.json")
 	j.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
-	return CreateJobSpecViaWeb(t, app, j)
+	return j
```

Comment:
Perhaps it's not an issue now, but having a job spec saved different to what's returned in memory will bite us. Can we fix this like we fixed the one above (`CreateHelloWorldJobViaWeb`)?

Reply:
Yeah I'm cleaning this one up, but it's a bit more complicated than it initially seems!